python_ta/__init__.py

```diff
@@ -117,45 +80,102 @@ def _check(module_name='', reporter=ColorReporter, number_of_messages=5, level='
     else:
         linter.read_config_file(os.path.join(os.path.dirname(__file__), '.pylintrc'))
     linter.load_config_file()
+    return linter
 
-    # Monkeypatch pylint
-    patch_all()
+
+def _verify_pre_check(filepath):
+    """Check student code for certain issues."""
 
     # Make sure the program doesn't crash for students.
     # Could use some improvement for better logging and error reporting.
     try:
         # Check for inline "pylint:" comment, which may indicate a student
         # trying to disable a check.
-        # TODO: Put this into a helper function.
-        with tokenize.open(spec.origin) as f:
+        with tokenize.open(filepath) as f:
             for (tok_type, content, _, _, _) in tokenize.generate_tokens(f.readline):
                 if tok_type != tokenize.COMMENT:
                     continue
                 match = pylint.utils.OPTION_RGX.search(content)
-                if match is None:
-                    continue
-                else:
-                    print('ERROR: string "pylint:" found in comment. No checks will be run.')
-                    return
+                if match is not None:
+                    print('ERROR: string "pylint:" found in comment. ' +
+                          'No check run on file `{}`\n'.format(filepath))
+                    return False
     except IndentationError as e:
         print('ERROR: python_ta could not check your code due to an ' +
               'indentation error at line {}'.format(e.lineno))
-        return
+        return False
     except tokenize.TokenError as e:
         print('ERROR: python_ta could not check your code due to a ' +
               'syntax error in your file')
+        return False
+    return True
+
+
+def _check(module_name='', reporter=ColorReporter, number_of_messages=5, level='all',
+           local_config_file='', pep8=False):
+    """Check a module for problems, printing a report.
+
+    <level> is used to specify which checks should be made.
+
+    The name of the module should be the name of a module,
+    or path(s) to Python file(s) or directory(ies).
+    """
+    valid_module_names = []
+
+    # Allow call to check with empty args
+    if module_name == '':
+        m = sys.modules['__main__']
+        spec = importlib.util.spec_from_file_location(m.__name__, m.__file__)
+        module_name = [spec.origin]
+    # Enforce API to expect 1 file or directory if type is list
+    elif isinstance(module_name, str):
+        module_name = list(filter(None, module_name.split(' ')))
```

Comment:
This is interesting. We should allow students to enter a path that contains spaces (students often have directories that contain spaces), so I think it's just better to remove this check.